diff --git a/mustache/mustache/src/Mustache/Compiler.php b/mustache/mustache/src/Mustache/Compiler.php
index 2a831d6..4df2952 100644
--- a/mustache/mustache/src/Mustache/Compiler.php
+++ b/mustache/mustache/src/Mustache/Compiler.php
@@ -520,7 +520,7 @@ class Mustache_Compiler
         $method  = $this->getFindMethod($id);
         $id      = ($method !== 'last') ? var_export($id, true) : '';
         $filters = $this->getFilters($filters, $level);
-        $value   = $escape ? $this->getEscape() : '$value';
+        $value   = $escape ? $this->getEscape() : 'is_array($value) ? \'\' : $value;';
 
         return sprintf($this->prepare(self::VARIABLE, $level), $method, $id, $filters, $this->flushIndent(), $value);
     }
@@ -599,7 +599,7 @@ class Mustache_Compiler
         return preg_replace("/\n( {8})?/", "\n" . str_repeat(' ', $bonus * 4), $text);
     }
 
-    const DEFAULT_ESCAPE = 'htmlspecialchars(%s, %s, %s)';
+    const DEFAULT_ESCAPE = '!is_array(%s) ? htmlspecialchars(%s, %s, %s) : \'\'';
     const CUSTOM_ESCAPE  = 'call_user_func($this->mustache->getEscape(), %s)';
 
     /**
@@ -615,7 +615,7 @@ class Mustache_Compiler
             return sprintf(self::CUSTOM_ESCAPE, $value);
         }
 
-        return sprintf(self::DEFAULT_ESCAPE, $value, var_export($this->entityFlags, true), var_export($this->charset, true));
+        return sprintf(self::DEFAULT_ESCAPE, $value, $value, var_export($this->entityFlags, true), var_export($this->charset, true));
     }
 
     /**
